var/spack/repos/builtin/packages/paraview/package.py

```diff
@@ -363,9 +360,41 @@ def nvariant_bool(feature):
         else:
             cmake_args.append('-DVTKm_ENABLE_CUDA:BOOL=%s' %
                               variant_bool('+cuda'))
-        if spec.satisfies('+cuda') and not spec.satisfies('cuda_arch=native'):
-            cmake_args.append('-DVTKm_CUDA_Architecture=%s' %
-                              spec.variants['cuda_arch'].value)
+
+        # VTK-m expects cuda_arch to be the arch name vs. the arch version.
+        if spec.satisfies('+cuda'):
+            supported_cuda_archs = {
+
+                # VTK-m and transitively ParaView does not support Tesla Arch
+                '20': 'fermi',
+                '21': 'fermi',
+                '30': 'kepler',
+                '32': 'kepler',
+                '35': 'kepler',
+                '37': 'kepler',
+                '50': 'maxwel',
+                '52': 'maxwel',
+                '53': 'maxwel',
+                '60': 'pascal',
+                '61': 'pascal',
+                '62': 'pascal',
+                '70': 'volta',
+                '72': 'volta',
+                '75': 'turing',
+                '80': 'ampere',
+                '86': 'ampere',
+                'none': 'native'
+            }
+
+            cuda_arch_value = 'native'
+            if 'cuda_arch' in spec:
+                cuda_arch_value = spec.variants['cuda_arch'].value
+                if cuda_arch_value not in supported_cuda_archs:
+                    raise InstallError("Uncompatible cuda_arch=%s" % cuda_arch_value)
+
```

Comment:
Here it's better to add conflicts instead of checking during installation. Adding conflicts sill make a wrong spec fail early.